source/common/http/header_map_impl.cc

```diff
@@ -214,38 +125,32 @@ void HeaderString::setCopy(absl::string_view view) {
 }
 
 void HeaderString::setInteger(uint64_t value) {
-  switch (type_) {
-  case Type::Reference: {
-    // Switch back to inline and fall through.
-    type_ = Type::Inline;
-    buffer_.dynamic_ = inline_buffer_;
-
-    FALLTHRU;
-  }
+  // Initialize the size to the max length, copy the actual data, and then
+  // reduce the size (but not the capacity) as needed
```

Comment:
Can you add a small note here about your perf experiment so that someone doesn't come along and ask the same question as I had? It's not intuitive to me that this is after versus an inline write, but I certainly believe your results.